### PR TITLE
docs: update tutorial and recording for runtime recording

### PR DESCRIPTION
**Pull request by hsgwa**
## Description
This PR updates the documentation by supporting the runtime recording feature.

For design update : #118 

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: docs/recording/recording.md [hunk 1]
```diff
@@ -77,7 +83,7 @@ Two terminals are needed for this method: One for starting a LTTng session, anot
 
 ## Starting LTTng session via ROS launch
 
-You can start LTTng session using ROS launch system. When you start a LTTng session in one terminal, you have to open another terminal for executing a target application as explained above. Operating multiple terminals is laborious for users. Launching LTTng session along with application by ROS launch is a reasonable way to apply CARET repeatedly.
+You can start LTTng session using ROS launch system. When you executing a target application in one terminal, you have to open another terminal for starting a LTTng session as explained above. Operating multiple terminals is laborious for users. Launching LTTng session along with application by ROS launch is a reasonable way to apply CARET repeatedly.
```
Comment (takam5f2) [inline docs/recording/recording.md:86]: I found an error on grammar.

```suggestion
You can start LTTng session using ROS launch system. When you have started a target application in one terminal, you have to open another terminal for starting a LTTng session as explained above. Operating multiple terminals is laborious for users. Launching LTTng session along with application by ROS launch is a reasonable way to apply CARET repeatedly.
```

**Review by takam5f2 — CHANGES_REQUESTED** [review]

File: docs/recording/cli_tool.md [hunk 2]
```diff
@@ -27,6 +27,21 @@ ros2 caret check_caret_rclcpp --workspace <path-to-workspace>
...
+
+This command runs LTTng session and recording sequence. ([See details](./recording.md))
+
+Recording begins after `All process started recording` is displayed.
+
+```bash
+ros2 caret record -s session_name -f 100 -v
+```
+
+Increasing the number of the -f option reduces the time required to start the recording.
```
[the remark this answers is not in the document]
Comment (hsgwa) [reply docs/recording/cli_tool.md]: addressed : [9ff9801](https://github.com/tier4/CARET_doc/pull/126/commits/9ff9801c5f46b47dbc0a5d62f4d5955ced367ff8)

File: docs/recording/recording.md [hunk 3]
```diff
@@ -2,32 +2,16 @@
...
-CARET uses LTTng for tracing a target application. A LTTng session has to be started before running the application. This page explains two different ways for it: Starting LTTng session manually and Starting LTTng session using ROS launch system.
+CARET uses LTTng for tracing a target application.
+This page explains two different ways for tracing: Starting LTTng session manually and Starting LTTng session using ROS launch system.
 
 Explanation in this page assumes CARET is installed to `~/ros2_caret_ws` and the sample application used in the tutorial section is located in `~/ros2_ws`.
 
 ## Starting LTTng session manually
 
-Two terminals are needed for this method: One for starting a LTTng session, another for running a target application.
+Two terminals are needed for this method: One for running a target application, another for starting a LTTng session.
```
[the remark this answers is not in the document]
Comment (hsgwa) [reply docs/recording/recording.md]: addressed : [9ff9801](https://github.com/tier4/CARET_doc/pull/126/commits/9ff9801c5f46b47dbc0a5d62f4d5955ced367ff8)

File: docs/recording/recording.md (hunk 1 above)
Comment (hsgwa) [reply docs/recording/recording.md]: addressed : [9ff9801](https://github.com/tier4/CARET_doc/pull/126/commits/9ff9801c5f46b47dbc0a5d62f4d5955ced367ff8)

File: docs/recording/recording.md [hunk 4]
```diff
@@ -61,13 +45,35 @@ Two terminals are needed for this method: One for starting a LTTng session, anot
...
 
-4. Press "Enter" key to stop the LTTng session in the terminal where the LTTng session runs
+4. Stop the target application
 
 <prettier-ignore-start>
 !!!info
-      A LTTng session needs to be started before running a target application. Otherwise, some trace points won't be recorded and analysis will fail later.
+      If you'd like to start measurement immediately after launch, reverse Step 1 and Step 2.
+      You can start the Lttng session first and start recording next as well.
+      However, if recording is started during launching an application, it may cause missing recordings.
```
[the remark this answers is not in the document]
Comment (hsgwa) [reply docs/recording/recording.md]: addressed : [9ff9801](https://github.com/tier4/CARET_doc/pull/126/commits/9ff9801c5f46b47dbc0a5d62f4d5955ced367ff8)

File: docs/recording/recording.md (hunk 1 above)
Comment (hsgwa) [reply docs/recording/recording.md]: Addressed [27b15ad](https://github.com/tier4/CARET_doc/pull/126/commits/27b15ad4a7158cf273770a8bd203a88c4c20af9a)

File: docs/recording/recording.md [hunk 5]
```diff
@@ -61,13 +45,35 @@ Two terminals are needed for this method: One for starting a LTTng session, anot
...
+      Please start recording either before launching or after launching is completed.
```
[the remark this answers is not in the document]
Comment (hsgwa) [reply docs/recording/recording.md]: Addressed [27b15ad](https://github.com/tier4/CARET_doc/pull/126/commits/27b15ad4a7158cf273770a8bd203a88c4c20af9a)

File: docs/recording/recording.md [hunk 6]
```diff
@@ -61,13 +45,35 @@ Two terminals are needed for this method: One for starting a LTTng session, anot
...
+3. Press "Enter" key to stop the LTTng session in the terminal where the LTTng session runs
 
-4. Press "Enter" key to stop the LTTng session in the terminal where the LTTng session runs
+4. Stop the target application
 
 <prettier-ignore-start>
 !!!info
-      A LTTng session needs to be started before running a target application. Otherwise, some trace points won't be recorded and analysis will fail later.
+      If you'd like to start measurement immediately after launch, reverse Step 1 and Step 2.
+      You can start the Lttng session first and start recording next as well.
```
[the remark this answers is not in the document]
Comment (hsgwa) [reply docs/recording/recording.md]: Addressed [27b15ad](https://github.com/tier4/CARET_doc/pull/126/commits/27b15ad4a7158cf273770a8bd203a88c4c20af9a)

File: docs/recording/recording.md [hunk 7]
```diff
@@ -61,13 +45,35 @@ Two terminals are needed for this method: One for starting a LTTng session, anot
      ros2 run caret_demos end_to_end_sample
      ```
 
-3. Stop the target application
+2. Open another terminal and start a LTTng session with the following commands
+
+   - (Optional) `ROS_TRACE_DIR` variable is a destination directory where recorded trace data will be stored. Default is `~/.ros/tracing`
```
[the remark this answers is not in the document]
Comment (hsgwa) [reply docs/recording/recording.md]: Addressed [27b15ad](https://github.com/tier4/CARET_doc/pull/126/commits/27b15ad4a7158cf273770a8bd203a88c4c20af9a)

File: docs/recording/cli_tool.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hsgwa) [reply docs/recording/cli_tool.md]: Addressed [27b15ad](https://github.com/tier4/CARET_doc/pull/126/commits/27b15ad4a7158cf273770a8bd203a88c4c20af9a)

**Review by takam5f2 — APPROVED** [review]
LTGM

**Pull request merged**